src/Compilers/Core/Portable/Symbols/IMethodSymbol.cs

```diff
@@ -263,6 +264,14 @@ public interface IMethodSymbol : ISymbol
         /// </summary>
         IMethodSymbol? PartialImplementationPart { get; }
 
+        /// <summary>
+        /// Returns the implementation attributes for the given method symbol.
+        /// </summary>
+        /// <remarks>
+        /// These attributes are not always emitted as attributes.
+        /// </remarks>
+        MethodImplAttributes ImplementationAttributes { get; }
```

Comment:
>ImplementationAttributes [](start = 29, length = 24)

Should we name the property "MethodImplAttributes"?

Reply:
@AlekseyTs It already exists with "ImplementationAttributes" name over the codebase.

Reply:
> It already exists with "ImplementationAttributes" name over the codebase.

That is an internal API and we aren't that picky about names of internal APIs. There are no concerns about if the name is going to confuse external consumers, etc. For example, it looks like reflection API is using ```GetMethodImplementationFlags``` name. Since we are returning the same information and using the same type, it is probably good to use terms already familiar to consumers. "MethodImplementationFlags", for example.

---
In reply to: [569594171](https://github.com/dotnet/roslyn/pull/50560#discussion_r569594171) [](ancestors = 569594171)